documents/resources.py

```diff
@@ -0,0 +1,22 @@
+import documents.state_machine as state_machine
+import utils
+from flask import jsonify, request, Blueprint
+
+documents = Blueprint("documents", __name__, template_folder="templates")
+
+@documents.route('', methods=['POST'])
+@utils.server_error_check
+@utils.authenticate_user
+def upsert_documents(payload):
+    user_id = payload['userid']
+    state_machine.get_user_by_user_id(user_id)
+    document_row = state_machine.upsert_document(user_id)
+    return jsonify(repr(document_row)), 200
+
+@documents.route('', methods=['GET'])
+@utils.server_error_check
```

Comment:
not authorizing user here?

Reply:
you should be. Right now you're allowing me to access all your documents by just knowing your user_id

Reply:
what you did is still not enough here @manthankumbhar 
You're checking if the user is authorized; what if the user is authorized, but uses a different user_id in the query params?

Reply:
okay, adding the check